process_reports.sh

```
@@ -0,0 +1,52 @@
+
+# Define the directory containing the reports
+REPORTS_DIR="reports"
+
+# Output file to store the results
+OUTPUT_FILE="plugin_evolution.csv"
+
+# Initialize the output file with headers
+echo "Date,Plugins_Without_Jenkinsfile,Plugins_With_Java8,Plugins_Without_Java_Versions" > $OUTPUT_FILE
+
+# Extract unique dates from filenames
+dates=$(ls $REPORTS_DIR | grep -oE '[0-9]{4}-[0-9]{2}-[0-9]{2}' | sort | uniq)
```

Shellcheck (0.10.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)